providers/pool.rb

```diff
@@ -232,6 +236,38 @@ def configure
       shell_out!(cmd)
       @new_resource.updated_by_last_action(true)
     end
+    if is_new_start_mode
+      was_updated = true
+      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
+      cmd << " /startMode:#{@new_resource.start_mode.to_s}"
+      Chef::Log.debug(cmd)
+      shell_out!(cmd)
+      @new_resource.updated_by_last_action(true)
+    end
+    if is_new_auto_start
+      was_updated = true
+      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
+      cmd << " /autoStart:#{@new_resource.auto_start.to_s}"
```

Comment:
Here we could use the `set config` syntax instead of `set apppool` in order to factorize the code.
```ruby
def update_simple_property(property, value)
  cmd = "#{appcmd(node)} set config /section:applicationPools"
  cmd << %{ "/[name='#{@new_resource.pool_name}'].#{property}:#{value}"}
  Chef::Log.debug(cmd)
  shell_out!(cmd)
  @new_resource.updated_by_last_action(true)
end
```
It will then become simplier to do that kind of thing:
```ruby
update_simple_property('autoStart', @new_resource.auto_start) if is_new_auto_start
...
update_simple_property('processModel.loadUserProfile', @new_resource.load_user_profile)
```
With some more refactoring, we could also handle complexe properties and and "is_new" check.